url_screenshot/icon_url_screenshot/actions/screenshot/action.py

```diff
@@ -0,0 +1,59 @@
+import komand
+from .schema import ScreenshotInput, ScreenshotOutput, Input, Output, Component
+# Custom imports below
+from komand.exceptions import PluginException
+import os
+import uuid
+import base64
+import time
+
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+
+
+class Screenshot(komand.Action):
+    CHROME_PATH = '/usr/bin/chromium'
+    CHROMEDRIVER_PATH = '/usr/bin/chromedriver'
+    WINDOW_SIZE = "1920,1080"
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='screenshot',
+                description=Component.DESCRIPTION,
+                input=ScreenshotInput(),
+                output=ScreenshotOutput())
+
+    def run(self, params={}):
+
+        name = f"/tmp/{str(uuid.uuid1())}.png"
+
+        chrome_options = Options()
+        chrome_options.add_argument("--headless")
+        chrome_options.add_argument("--window-size=%s" % self.WINDOW_SIZE)
+        chrome_options.add_argument("--no-sandbox")
+        chrome_options.binary_location = self.CHROME_PATH
+
+        url = params.get(Input.URL, "")
+        delay = params.get(Input.DELAY, 0)
+        if not url.lower().startswith('http'):
+            raise PluginException(cause='Input Error:', assistance='URLs need to start with "http"')
+        driver = webdriver.Chrome(executable_path=self.CHROMEDRIVER_PATH, options=chrome_options)
+        driver.get(url)
+        time.sleep(delay)
```

Comment:
Looks like you should be using WebDriverWait https://stackoverflow.com/questions/26566799/wait-until-page-is-loaded-with-selenium-webdriver-for-python

Reply:
Reviewing the stack overflow link, I don't believe so.  The webdriver.wait is for waiting for a specific element to load in response to an ajax call or similar.  The webdriver.get(url) will wait until the page is loaded.

The delay here is meant to be controllable so as to catch the output of scripts which may be triggering at the completion of loading the page.  Some pages look clean when you first land on them and then, 5 seconds later, they look awful.  This delay is meant to catch the later state.